masterbase/app.py

```diff
@@ -140,6 +141,15 @@ async def demodata(request: Request, api_key: str, session_id: str) -> Stream:
     return Stream(bytestream_generator, media_type=MediaType.TEXT, headers=headers)
 
 
+@post("/report", guards=[valid_key_guard])
+async def report_player(request: Request, api_key: str, session_id: str, target_steam_id: str):
+    """Add a player report."""
+    # TODO: currently performs no session verification or Steam ID validation/cross-verification.
+    engine = request.app.state.engine
+    add_report(engine, session_id, target_steam_id)
+    {"report_added": True}
```

Comment:
possible that the `add_report` throws an error (I would bet an `IntegrityError` but worth checking, catching, and returning a graceful 401

Reply:
@kavorite like try inserting into reports with a session ID that doesnt exist in demo_sessions table

Reply:
I checked out your branch and ran migrations:
```sh
pdm run alembic upgrade head                         

INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 58fb39990d30 -> 405672cf4046, reports
```

Then lets connect:
```sh
pgcli postgresql://MEGASCATTERBOMB:masterbase@localhost:8050/demos
Server: PostgreSQL 16.1 (Debian 16.1-1.pgdg110+1)
Version: 4.0.1
Home: http://pgcli.com
demos>
```

Now find valid `session_id`:
```
demos> select session_id from demo_sessions
+-----------------------------------------+
| session_id                              |
|-----------------------------------------|
| 242328518628336623079943370902371134809 |
| 214609055706631507751012305309291754995 |
| 242672456335999900255056495522936833092 |
| 297739075442803444840938830259614456160 |
| 4096009816984305196750269377751802420   |
| 113475737134310454011779855209743437992 |
+-----------------------------------------+
SELECT 6
Time: 0.014s
```

See that reports is empty:
```
demos> select * from reports
+------------+-----------------+------------+
| session_id | target_steam_id | created_at |
|------------+-----------------+------------|
+------------+-----------------+------------+
SELECT 0
Time: 0.007s
```

Now try to insert:
```
demos> insert into reports values (1, 2, NOW())
insert or update on table "reports" violates foreign key constraint "reports_session_id_fkey"
DETAIL:  Key (session_id)=(1) is not present in table "demo_sessions".
Time: 0.008s
```

On the sqlalchemy side, this will raise an `IntegrityError` which we can either catch and raise, or add a guard that ensures the session_id the user api is receiving is valid

Reply:
Ah, looks like errors get more specific, it actually raises a `psycopg2.errors.ForeignKeyViolation` on the python side

Reply:
The full error is `sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation)` so catching a `sqlalchemy.exc.IntegrityError` should suffice if you decide to go that route

Reply:
You're sure retroactive ingestion per [mega's request](https://github.com/MegaAntiCheat/masterbase/pull/55#issuecomment-2126636182) can take place if we enforce `valid_session_guard` after merging https://github.com/MegaAntiCheat/masterbase/pull/51? I'm thinking of just guarding it, but if `session_id` is based on the server where an accusation is taking place, then the client needs to have already persisted that at some point in order for all this to actually work (cc @Bash-09)

Reply:
I am suggesting a new guard that just checks that the session_id exists in the demo_sessions table, but I think we can rely on the `sqlalchemy.exc.IntegrityError` and ask for forgiveness

Reply:
You're right. There's no need to complicate things further than try/catch. Committing: (https://github.com/MegaAntiCheat/masterbase/pull/55/commits/3ffcc44eb1d5f359c75a0f656770673b1e06f87b)